decidim-core/app/models/decidim/action_log.rb

```diff
@@ -216,14 +216,20 @@ def self.lazy_relation(id_method, klass_name, cache)
     # Whether this activity or log is visible for a given user (can also be nil)
     #
     # Returns a True/False.
+    # rubocop:disable Metrics/CyclomaticComplexity
     def visible_for?(user)
-      return false if resource_lazy.blank?
-      return false if participatory_space_lazy.blank?
-      return false if resource_lazy.respond_to?(:deleted?) && resource_lazy.deleted?
-      return false if resource_lazy.respond_to?(:hidden?) && resource_lazy.hidden?
-      return false if resource_lazy.respond_to?(:can_participate?) && !resource_lazy.can_participate?(user)
-
+      begin
+        return false if resource_lazy.blank?
+        return false if participatory_space_lazy.blank?
+        return false if resource_lazy.respond_to?(:deleted?) && resource_lazy.deleted?
+        return false if resource_lazy.respond_to?(:hidden?) && resource_lazy.hidden?
+        return false if resource_lazy.respond_to?(:can_participate?) && !resource_lazy.can_participate?(user)
+      rescue NameError => e
```

Comment:
Why not add the rescue block to the method itself?

Reply:
You're right! done.